src/account/UpgradeableModularAccount.sol

```diff
@@ -680,30 +725,16 @@ contract UpgradeableModularAccount is
         }
     }
 
-    function _globalValidationAllowed(bytes4 selector) internal view returns (bool) {
-        if (
-            selector == this.execute.selector || selector == this.executeBatch.selector
-                || selector == this.installExecution.selector || selector == this.uninstallExecution.selector
-                || selector == this.installValidation.selector || selector == this.uninstallValidation.selector
-                || selector == this.upgradeToAndCall.selector
-        ) {
-            return true;
-        }
-
-        return getAccountStorage().executionData[selector].allowGlobalValidation;
-    }
-
     function _checkIfValidationAppliesSelector(bytes4 selector, ModuleEntity validationFunction, bool isGlobal)
         internal
         view
     {
-        AccountStorage storage _storage = getAccountStorage();
-
         // Check that the provided validation function is applicable to the selector
         if (isGlobal) {
-            if (!_globalValidationAllowed(selector) || !_storage.validationData[validationFunction].isGlobal) {
-                revert ValidationFunctionMissing(selector);
+            if (_globalValidationAllowed(selector) && _isValidationGlobal(validationFunction)) {
+                return;
             }
+            revert ValidationFunctionMissing(selector);
```

Comment:
Nit: stylistically, I prefer the format of `if (...) { revert ...; }` instead of adding the return line here. Wdyt?

Reply:
I tend to prefer early returns because it shows exactly where the positive path ends, if we put the revert in the "if" scope, then the positive path continues to the end of the function, which means we have to read through the remainder of the function body.

In this case, the function body is totally empty and your approach saves a line (and is literally the approach in the parent "else" block) so addressed in [12fc8ad](https://github.com/erc6900/reference-implementation/pull/123/commits/12fc8ad8169778a419ac730ac48d6271c94e5ecd)!